core/src/main/scala/caliban/schema/Types.scala

```diff
@@ -130,6 +130,18 @@ object Types {
         t.possibleTypes.getOrElse(Nil).foldLeft(list2) { case (types, subtype) => collectTypes(subtype, types) }
     }
 
+  def unify(l: List[__Type]): Option[__Type] =
```

Comment:
Maybe add a comment on what he does because it's quite specific.

Reply:
Added in 5756594.